[ADD] crm_l10n_eu_nace: NACE for leads

yajo opened the pull request:
This module adds the concept of NACE activity to the CRM main objects (lead,
opportunity), and propagates them through the workflow.

Depends on:

- [x] https://github.com/OCA/community-data-files/pull/53
- [x] https://github.com/OCA/community-data-files/pull/90

@Tecnativa TT17645

pedrobaeza commented on crm_l10n_eu_nace/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,4 @@
+* Pedro M. Baeza (`Tecnativa <https://www.tecnativa.com/>`__)
+* Antonio Espinosa (`Tecnativa <https://www.tecnativa.com/>`__)
+* David Vidal (`Tecnativa <https://www.tecnativa.com/>`__)
+* Jairo Llopis (`Tecnativa <https://www.tecnativa.com/>`__)
```
Better to use 2 indentation levels as we do in other modules
```suggestion
* `Tecnativa <https://www.tecnativa.com/>`__:

  * Pedro M. Baeza
  * Antonio Espinosa
  * David Vidal
  * Jairo Llopis
```

pedrobaeza commented on crm_l10n_eu_nace/README.rst:
```diff
@@ -0,0 +1,77 @@
+.. contents::
+   :local:
+
+Bug Tracker
```
Add `INSTALL.rst` saying that it depends on `l10n_eu_nace` module in OCA/community-data-files.

yajo replied:
All fixed, thanks.

pedrobaeza commented on crm_l10n_eu_nace/models/crm_lead.py:
```diff
@@ -0,0 +1,24 @@
+            result["nace_id"] = self.nace_id.ids
+        if self.secondary_nace_ids:
+            result["secondary_nace_ids"] = [
+                (4, self.secondary_nace_ids.ids, 0)
```
```suggestion
                (4, x) for x in self.secondary_nace_ids.ids
```

yajo replied:
Don't you always need the 3rd element in the tuple?

pedrobaeza replied:
Not really, but that's not the main problem here: you have to add a `(4, ...` command for each element.

pedrobaeza replied:
This is still not changed

pedrobaeza commented on crm_l10n_eu_nace/tests/test_crm_nace.py:
```diff
@@ -0,0 +1,34 @@
+                "partner_name": "someone",
+                "nace_id": self.nace_0.id,
+                "secondary_nace_ids": [
+                    (4, (self.nace_1.id, self.nace_2.id), 0)],
```
```suggestion
                    (4, self.nace_1.id), (4, self.nace_2.id)],
```

pedrobaeza commented on crm_l10n_eu_nace/tests/test_crm_nace.py:
```diff
@@ -0,0 +1,34 @@
+        lead.handle_partner_assignation()
+        self.assertEqual(lead.partner_id.nace_id, self.nace_0)
+        self.assertEqual(
+            lead.partner_id.secondary_nace_id, self.nace_1 | self.nace_2)
```
```suggestion
            lead.partner_id.secondary_nace_ids, self.nace_1 | self.nace_2)
```

Commit:
[ADD] crm_l10n_eu_nace: NACE for leads

This module adds the concept of NACE activity to the CRM main objects (lead,
opportunity), and propagates them through the workflow.

@Tecnativa TT17645

Apply suggestions from code review

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

HaraldPanten reviewed (approved):
LGTM 👍

pedrobaeza commented:
/ocabot merge nobump
